test/runner.js

```diff
@@ -0,0 +1,192 @@
+'use strict'
+
+let fs = require('fs')
+let path = require('path')
+let expect = require('chai').expect
+let rimraf = require('rimraf')
+let Launcher = require('webdriverio/build/lib/launcher')
+let parseXmlString = require('xml2js').parseString
+
+let configFile = './test/fixtures/wdio.conf.js'
+let resultsDir = path.join(__dirname, '../allure-results') 
+
+describe('"before all" hook', () => {
+
+  beforeEach(() => {
+    clean()
+  })
+
+  it('should not show up in the results when it is passing', () => {
+
+    return run(['before-all-passing']).then((code) => {
+
+      return getResultsXML()
+        .then((results) => {
+
+          expect(results).to.have.lengthOf(1)
+          expect(results[0]['ns2:test-suite']['test-cases']).to.have.lengthOf(1)
+          
+          results[0]['ns2:test-suite']['test-cases'][0]['test-case']
+            .forEach((testCase) => {
+
+              switch(testCase.name[0]) {
+
+                case '"before" hook':
+                case '"before all" hook':
+                case '"before each" hook':
+                case '"after all" hook':
+                case '"after each" hook':
+                  throw new Error('\'' + testCase.name + '\' was not expected in the results when no hook fails')
+                  break
+
+              }
+
+            })
```

Comment:
Since we have assertion library, you should use it to make checks. It is especially important with Allure, because Allure by design has different statuses for AssertionError (failed) and other types (broken). 


You can do this check with matchers in the following way:

```js
const hookNames = ['"before" hook', '"before all" hook', '"before each" hook', '"after all" hook', '"after each" hook']
const hookErrors = results[0]['ns2:test-suite']['test-cases'][0]['test-case'].filter(testCase => hookNames.indexOf(testCase.name[0]) > -1)
expect(hookErrors).to.be.empty()
```